hnn_core/viz.py

```diff
@@ -836,8 +836,41 @@ def plot_connectivity_matrix(net, conn_idx, ax=None, show_weight=True,
     return ax.get_figure()
 
 
-def plot_cell_connectivity(net, conn_idx, src_gid, ax=None, colorbar=True,
-                           colormap='viridis', show=True):
+def _update_target_plot(ax, conn, src_gid, src_type_pos, target_type_pos,
+                        src_range, target_range, nc_dict, colormap):
+    from .cell import _get_gaussian_connection
+
+    # Extract indeces to get position in network
+    # Index in gid range aligns with net.pos_dict
+    target_src_pair = conn['gid_pairs'][src_gid]
+    target_indeces = np.where(np.in1d(target_range, target_src_pair))[0]
```

Comment:
Great solution, much cleaner

Reply:
it's your code, I didn't do anything ... just moved it around :)